src/NBitcoin/Transaction.cs

```diff
@@ -1111,10 +1111,12 @@ internal void ReadWrite(BitcoinStream stream)
 
     //https://en.bitcoin.it/wiki/Transactions
     //https://en.bitcoin.it/wiki/Protocol_specification
-    public class Transaction : IBitcoinSerializable
+    public class Transaction : IBitcoinSerializable, IHaveNetworkOptions
```

Comment:
I really don't like this approach, to put the options in trx.
I would rather pass options in to methods they are needed in but I appreciate this is hard to do it otherwise right now.

Did we not agree we will not put options in the entities themselves?

Reply:
It would be extremely hard to do. Due to the amount of code involved it is already looking like a bad idea to have separated the options from the block. It helps to see the options as something from which we derive serialization rather than seeing this as mixing serialization into the object.

Reply:
The ToString is decorative its not essential for logic.

Reply:
hmm christ the problem this is entangled in so many parts of the code.

Reply:
We have to pass the Network (or representative) object every where it is needed or we have to piggy-back that information off existing objects. It depends on how much code we are willing to change. 

Simply passing the Network object around does have some drawback. For one the custom parent objects would be unable to influence their child objects via the Network object. In this sense the NetworkOptions and actually recording that on each object does appear to have some advantages. Although ToString is just cosmetic it does help developers to be able to see the hash on a block.